### PR TITLE
:zap: Improve performance of `ActiveIndexedComponent`

### PR DESCRIPTION
**Pull request by ruaridhw**
- Iterate over `self._data` directly

## Summary/Motivation:
There methods can be substantially sped up by iterating directly over `_data`

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by mhusbyn — APPROVED** [review]
Looks fine to me. Is there a PR to the main Pyomo project for this as well? I'm assuming we also want it there right?

EDIT: All the pipelines are failing. That's probably an issue?

File: pyomo/core/base/indexed_component.py [hunk 1]
```diff
@@ -803,13 +803,23 @@ def activate(self):
         """Set the active attribute to True"""
         super(ActiveIndexedComponent, self).activate()
         if self.is_indexed():
-            for component_data in itervalues(self):
+            try:
+                component_data_iter = itervalues(self._data)
+            except AttributeError:
+                component_data_iter = iter(self._data)
```
Comment (mhusbyn) [inline pyomo/core/base/indexed_component.py:809]: Could you add a comment or something here now? It's unclear what attribute it's erroring on.

**Comment by ruaridhw** [comment]
> Looks fine to me. Is there a PR to the main Pyomo project for this as well? I'm assuming we also want it there right?

Could do. I think that's lower priority though so I won't worry about it for now.

> EDIT: All the pipelines are failing. That's probably an issue?

Pipelines are passing now 🎉  (Except for 3.8 which was failing previously...)

**Comment by mhusbyn** [comment]
> Could do. I think that's lower priority though so I won't worry about it for now.

When would we do it if not now? To be fair, I don't know how much hassle it is to do. I can just imagine them doing some refactor of this area of code and we have to spend a lot of time on merge conflicts, etc. I'll leave it up to your discretion, but the less our codebase diverges from Pyomo's, the less pain and time we'll have maintaining it.

**Pull request merged**